### PR TITLE
convert db lib to use slog, part 1

### PR DESCRIPTION
**Pull request by greedy52**
part 1, converted all proxy servers, and main db server.

### CODE REVIEW

File: lib/srv/db/server.go [hunk 1]
```diff
@@ -296,8 +297,11 @@ type Server struct {
 	reconcileCh chan struct{}
 	// mu protects access to server infos and databases.
 	mu sync.RWMutex
-	// log is used for logging.
-	log *logrus.Entry
+	// logrusLogger is used for logging.
+	// TODO (greedy52) delete when possible.
+	logrusLogger *logrus.Entry
```
Comment (Tener) [inline lib/srv/db/server.go:302]: Deprecate the field? Rename to `deprecatedLogger`?

**Review by Tener — APPROVED** [review]

**Review by flyinghermit — APPROVED** [review]
Looks good, I wonder if we can create base logger `dbLogger := slog.With(..dbname, dbtype )` with db name and db type so they are only formatted and evaluated once.

**Comment by greedy52** [comment]
> Looks good, I wonder if we can create base logger `dbLogger := slog.With(..dbname, dbtype )` with db name and db type so they are only formatted and evaluated once.

Not for the db server but we are doing that for server context:
https://github.com/gravitational/teleport/blob/51ce965c9af7366b8815b0a71883b72f61a4421f/lib/srv/db/server.go#L1169-L1172

We were actually discussing this on @Tener 's other change if we want to add more fields here. @Tener what do you think?

**Comment by Tener** [comment]
> We were actually discussing this on @Tener 's other change if we want to add more fields here. @Tener what do you think?

I don't feel strongly either way. On one hand, adding these fields may be helpful in some contexts. On the other, the database-specific handling is done in various engines, and these already include these fields, so it probably doesn't change available information.

**Comment by greedy52** [comment]
Friendly ping @zmb3

**Review by zmb3 — APPROVED** [review]

**Pull request merged**